webknossos-datastore/app/com/scalableminds/webknossos/datastore/dataformats/wkw/WKWDataFormat.scala

```diff
@@ -39,17 +39,39 @@ object WKWDataFormat extends DataSourceImporter with WKWDataFormatHelper {
     }
   }
 
-  private def exploreResolutions(baseDir: Path, previous: Option[List[Point3D]])(implicit report: DataSourceImportReport[Path]): Box[List[(WKWHeader, Either[Int, Point3D])]] = {
-    def resolutionDirFilter(path: Path): Boolean = path.getFileName.toString.toIntOpt.isDefined
-    def resolutionDirInt(path: Path): Int = path.getFileName.toString.toInt
+  private def exploreResolutions(baseDir: Path)(implicit report: DataSourceImportReport[Path]): Box[List[(WKWHeader, Either[Int, Point3D])]] = {
+
+    def parseResolutionName(path: Path): Option[Either[Int, Point3D]] = {
+        path.getFileName.toString.toIntOpt match {
```

Comment:
You can use regexs in pattern matching (http://www.scala-lang.org/api/2.12.1/scala/util/matching/Regex.html) so there is no need for `try` `catch`.